ipn/ipnlocal/c2n.go

```diff
@@ -151,6 +159,22 @@ func (b *LocalBackend) handleC2NUpdate(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// Check if update was already started, and mark as started.
+	b.mu.Lock()
```

Comment:
Consider maybe pulling this into a method, e.g., 

```go
if !testAndSetC2NUPdateStarted(true) {
   res.Err = "not supported"
   return
}
defer func() { /* … */ }()
```

where the method checks whether it's already set while holding the lock and reports back.